server/server.go

```diff
@@ -532,7 +535,7 @@ func (s *Server) getTLSConfig() *tls.Config {
 
 func killConn(conn *clientConn) {
 	sessVars := conn.ctx.GetSessionVars()
-	atomic.CompareAndSwapUint32(&sessVars.Killed, 0, 1)
+	atomic.StoreUint32(&sessVars.Killed, 1)
```

Comment:
Is there some tricky reason that we use `CAS` instead of `Store` here at the begging? Please help us to confirm this. @tiancaiamao

Reply:
Exception handling is a hard.

If we use `CAS`, exception handling is the same as error handling:
- one of the goroutine detect kill (CAS), it return an error
- the error is throw to the main loop
- the main loop handle worker's Close

This processing is exactly the same as the error handling, when one of the worker get an error.

Exception handling is much difficult than error handling.
- The the worker and main loop does not know when the exception happen
- Every worker has to handle the exception
- The exception handle is tricky, should the worker kill itself or report error to the main loop and let it kill worker? Will the Close repeat message repeat and Close multiple times?